akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -1104,6 +1104,7 @@ abstract class ShardCoordinator(
                 evt.region)
 
               sendHostShardMsg(evt.shard, evt.region)
+              // FIXME: shard region may ignore the HostShard message
               getShardHomeSender ! ShardHome(evt.shard, evt.region)
```

Comment:
We are very optimistic about this. We send the `HostShard` and immediately let the requester send messages to the region. 

This will probably work in most of the cases, but we could alternatively pass the requester in the `HostShard` message and let the region tell the news. 

The problem is what to do when it ignores the message.

Reply:
Could this be what will happen?

1. A requests shard 1 
2. coordinator decides it will be on node B
3. coordinator tells A that region B hosts shard, and commands B to host it
4. B ignores the command to host
5. A sends a message for shard 1 to B
6. If region B is still alive it will buffer it and ask coordinator where shard 1 is
7. Region B is hindered from shutting down while there are buffered messages
8. Coordinator has not seen B terminate so will say 1 is on B

Reply:
yeah, certainly one of the possible scenarios. 

And will the the region B deliver the message to the shard if it previously decided not to host it? I don't think so. Looking into that right now, I will bring some news. Stay tuned...

Reply:
ok, it will deliver to the shard, but...

If will first need to initialise it and only on the next `Retry` polling, it will deliver the buffered messages. The normal retry polling is 2 seconds. So, quite long.

Reply:
for the record, it won't wait for the next `Retry` polling. It will deliver the message as soon as the shard notifies the region that it started.